app/src/main/java/com/alphawallet/app/entity/SignaturePair.java

```diff
@@ -57,13 +61,25 @@ public String formQRMessage()
     }
 
     //should only be one by one for now
+    /**
+     * Note: all values must be in decimal (base 10) to allow the QR encoding optimiser to work most efficiently
+     *
+     * Format is:
+     * one decimal digit for future proofing in case we add to this format
+     * two decimal digits giving length of the decimal encoded token ID
+     * remaining digits are the signature in decimal
+     *
+     */
     public static String generateSelection721Tickets(List<BigInteger> tokenIds) {
+        String prefix = "0"; //future proof the format - this is one digit to refer to token count or protocol format
         StringBuilder stringBuilder = new StringBuilder();
         for(BigInteger token: tokenIds) {
-            stringBuilder.append(Numeric.toHexStringNoPrefix(token));
+            stringBuilder.append(token.toString(10));
             stringBuilder.append(",");
         }
-        return stringBuilder.substring(0, stringBuilder.length() - 1);
+        String tokenIdsString = stringBuilder.substring(0, stringBuilder.length() - 1);
```

Comment:
@JamesSmartCell Stringbuilder.toString?

Reply:
I don't understand what this comment means.

Reply:
Why use substring when you can just use the string directly? I.e. you are using substring for the full length so there is no need to substring

Reply:
I did minimum of changes to the code you added ... I guessed that you did the substring because eventually it could handle multiple tokens.